pipeline/trusted-resources/pkg/trustedtask/sign_test.go

```diff
@@ -0,0 +1,194 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package trustedtask
+
+import (
+	"context"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/google/go-containerregistry/pkg/authn/k8schain"
+	"github.com/google/go-containerregistry/pkg/registry"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/sigstore/sigstore/pkg/signature"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	fakek8s "k8s.io/client-go/kubernetes/fake"
+)
+
+func TestSignTaskSpec(t *testing.T) {
+	sv, err := getSignerVerifier(password)
+	if err != nil {
+		t.Errorf("failed to get signerverifier %v", err)
+	}
+
+	tcs := []struct {
+		name     string
+		signer   signature.SignerVerifier
+		taskSpec *v1beta1.TaskSpec
+		wantErr  bool
+	}{{
+		name:     "Sign TaskSpec",
+		signer:   sv,
+		taskSpec: taskSpecTest,
+		wantErr:  false,
+	}, {
+		name:     "Empty TaskSpec",
+		signer:   sv,
+		taskSpec: nil,
+		wantErr:  false,
+	}, {
+		name:     "Empty Signer",
+		signer:   nil,
+		taskSpec: taskSpecTest,
+		wantErr:  true,
+	},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			if _, err := SignTaskSpec(tc.signer, taskSpecTest); (err != nil) != tc.wantErr {
+				t.Errorf("SignTaskSpec() get err %v, wantErr %t", err, tc.wantErr)
+			}
+		})
+	}
+
+}
+
+func TestSignRawPayload(t *testing.T) {
+	sv, err := getSignerVerifier(password)
+	if err != nil {
+		t.Errorf("failed to get signerverifier %v", err)
+	}
+
+	tcs := []struct {
+		name    string
+		signer  signature.SignerVerifier
+		payload []byte
+		wantErr bool
+	}{{
+		name:    "Sign raw payload",
+		signer:  sv,
+		payload: []byte("payload"),
+		wantErr: false,
+	}, {
+		name:    "Empty payload",
+		signer:  sv,
+		payload: nil,
+		wantErr: false,
+	}, {
+		name:    "Empty Signer",
+		signer:  nil,
+		payload: []byte("payload"),
+		wantErr: true,
+	},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			if _, err := SignRawPayload(tc.signer, tc.payload); (err != nil) != tc.wantErr {
+				t.Errorf("SignRawPayload() get err %v, wantErr %t", err, tc.wantErr)
+			}
+		})
+	}
+
+}
+
+func TestDigest(t *testing.T) {
+	ctx := context.Background()
+
+	k8sclient := fakek8s.NewSimpleClientset(sa)
+
+	// Create registry server
+	s := httptest.NewServer(registry.New())
+	defer s.Close()
+	u, _ := url.Parse(s.URL)
+
+	// Push OCI bundle
+	if _, err := pushOCIImage(t, u, ts); err != nil {
+		t.Error(err)
+	}
+
+	kc, err := k8schain.New(ctx, k8sclient, k8schain.Options{
```

Comment:
I think we can omit the keychain and just use k8schain.NewNoClient - we're not really trying to test auth here, and the sign cmd shows we don't actually need this to function properly.

Reply:
yes